dask/array_api/_statistical_functions.py

```diff
@@ -0,0 +1,52 @@
+# Statistical Functions
+# https://data-apis.org/array-api/latest/API_specification/statistical_functions.html
+
+import dask.array as da
+
+from ._dtypes import _floating_dtypes, _numeric_dtypes, float32, float64
+
+
+def max(x, /, *, axis=None, keepdims=False):
+    if x.dtype not in _numeric_dtypes:
+        raise TypeError("Only numeric dtypes are allowed in max")
+    return da.max(x, axis=axis, keepdims=keepdims)
+
+
+def mean(x, /, *, axis=None, keepdims=False):
+    if x.dtype not in _floating_dtypes:
+        raise TypeError("Only floating-point dtypes are allowed in mean")
```

Comment:
FYI, this sort of thing doesn't hurt to have, but it also isn't required by the spec. Type combinations outside of those that say "should" in the spec are just considered out of scope. The NumPy implementation is particularly strict and disallows anything not explicitly required by the spec, but it is not required that other implementations do this. The test suite will only check for required combinations (i.e., this line won't even be covered by array-api-tests). There are some cases where an exception is required by the spec, and those are covered by the test suite. 

OTOH, mixed-type promotion and cross-type casting in particular are out of scope in the spec specifically because they are considered bad for various reasons. So it's also not so bad to have this sort of thing in there just to make the API cleaner.

Reply:
That's good to know. I was following the NumPy implementation and being strict. In general, it's easier to start out strict, and loosen an implementation rather than the other way round - due to backwards compatibility guarantees. However, this is probably an area where we might want to just use the existing Dask implementation. Interested to see what other @dask/array folks think?

BTW I wonder if the array API spec should use capitalised "SHOULD" rather than "should" to make it a bit clearer it's using RFC 2119 levels?